rfcs/0049-flakes.md

```diff
@@ -137,28 +137,24 @@ installable derivation) and a NixOS module.
 
 ```
 {
-  name = "dwarffs";
-
-  edition = 201906;
+  edition = 201909;
 
   description = "A filesystem that fetches DWARF debug info from the Internet on demand";
 
-  inputs = [ "nixpkgs" ];
-
-  outputs = inputs: rec {
+  outputs = { self, nixpkgs }: rec {
```

Comment:
Should the set be recursive? Just read yesterday about how recursive overlays should be avoided and the proper way is to use the self fixpoint.

Reply:
No, it doesn't have to be recursive, you can also use `self` to refer to other outputs.

Reply:
So the [advice about overlays](https://blog.flyingcircus.io/2017/11/07/nixos-the-dos-and-donts-of-nixpkgs-overlays/), specifically about a "recursive attrset", doesn't apply?

Reply:
Yes, because this is not an overlay and there is no override mechanism at the moment (so the `rec` is the same as `self.outputs`).